test/integ/conftest.py

```diff
@@ -1,87 +1,281 @@
+    queue_b: Queue = field(init=False)
+    fleet: Fleet = field(init=False)
+    scaling_queue: Queue = field(init=False)
+    scaling_fleet: Fleet = field(init=False)
 
+    farm_id: InitVar[str]
+    queue_a_id: InitVar[str]
+    queue_b_id: InitVar[str]
+    fleet_id: InitVar[str]
+    scaling_queue_id: InitVar[str]
+    scaling_fleet_id: InitVar[str]
 
-@pytest.fixture(scope="session")
-def queue(deadline_resources: DeadlineResources) -> Queue:
-    return deadline_resources.queue
+    def __post_init__(
+        self,
+        farm_id: str,
+        queue_a_id: str,
+        queue_b_id: str,
+        fleet_id: str,
+        scaling_queue_id: str,
+        scaling_fleet_id: str,
+    ) -> None:
+        object.__setattr__(self, "farm", Farm(id=farm_id))
+        object.__setattr__(self, "queue_a", Queue(id=queue_a_id, farm=self.farm))
+        object.__setattr__(self, "queue_b", Queue(id=queue_b_id, farm=self.farm))
+        object.__setattr__(self, "fleet", Fleet(id=fleet_id, farm=self.farm))
+        object.__setattr__(self, "scaling_queue", Queue(id=scaling_queue_id, farm=self.farm))
+        object.__setattr__(self, "scaling_fleet", Fleet(id=scaling_fleet_id, farm=self.farm))
 
 
 @pytest.fixture(scope="session")
-def fleet(deadline_resources: DeadlineResources) -> Fleet:
-    return deadline_resources.fleet
+def deadline_resources() -> Generator[DeadlineResources, None, None]:
+    """
+    Gets Deadline resources required for running tests.
 
+    Environment Variables:
+        FARM_ID: ID of the Deadline farm to use.
+        QUEUE_A_ID: ID of a non scaling Deadline queue to use for tests.
+        QUEUE_B_ID: ID of a non scaling Deadline queue to use for tests.
+        FLEET_ID: ID of a non scaling Deadline fleet to use for tests.
+        SCALING_QUEUE_ID: ID of the Deadline scaling queue to use.
+        SCALING_FLEET_ID: ID of the Deadline scaling fleet to use.
 
-@pytest.fixture(scope="session")
-def job_run_as_user() -> PosixSessionUser:
-    return PosixSessionUser(
-        user="job-run-as-user",
-        group="job-run-as-user-group",
+    Returns:
+        DeadlineResources: The Deadline resources used for tests
+    """
+    farm_id = os.environ["FARM_ID"]
+    queue_a_id = os.environ["QUEUE_A_ID"]
+    queue_b_id = os.environ["QUEUE_B_ID"]
+    fleet_id = os.environ["FLEET_ID"]
+
+    scaling_queue_id = os.environ["SCALING_QUEUE_ID"]
+    scaling_fleet_id = os.environ["SCALING_FLEET_ID"]
+
+    LOG.info(
+        f"Configured Deadline Cloud Resources, farm: {farm_id}, scaling_fleet: {scaling_fleet_id}, scaling_queue: {scaling_queue_id} ,queue_a: {queue_a_id}, queue_b: {queue_b_id}, fleet: {fleet_id}"
+    )
+
+    yield DeadlineResources(
+        farm_id=farm_id,
+        queue_a_id=queue_a_id,
+        queue_b_id=queue_b_id,
+        fleet_id=fleet_id,
+        scaling_queue_id=scaling_queue_id,
+        scaling_fleet_id=scaling_fleet_id,
     )
 
 
 @pytest.fixture(scope="session")
 def worker_config(
```

Comment:
And this one? https://github.com/aws-deadline/deadline-cloud-test-fixtures/blob/mainline/src/deadline_test_fixtures/fixtures.py#L381